vm/devices/tpm/src/ak_cert.rs

```diff
@@ -19,6 +19,20 @@ pub enum TpmAkCertType {
     HwAttested(Box<dyn RequestAkCert>),
 }
 
+impl TpmAkCertType {
+    /// Get the `RequestAkCert` from the enum
+    ///
+    /// The function must only be called if the enum is [`TpmAkCertType::HwAttested`]
```

Comment:
No need to make this policy choice at the helper-level. Return an option, and let the caller unwrap() and/or handle the None case however they like